p2p/conn/connection_test.go

```diff
@@ -769,3 +770,273 @@ func stopAll(t *testing.T, stoppers ...stopper) func() {
+			// Ideal transmission time for 50 messages is ~1 second.
+			// Increasing SendQueueCapacity should not affect transmission
+			// delay.
+			name: "queue capacity = 100, " +
+				"total load = 50 KB, " +
+				"traffic rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          1 * 50,
+			sendQueueCapacity: 100,
+			messagingRate:     20 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 4
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// The test runs for 100 messages, expecting a total transmission time of ~2 seconds.
+			name: "queue capacity = 100, " +
+				"total load = 2 * 50 KB, " +
+				"traffic rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          2 * 50,
+			sendQueueCapacity: 10,
+			messagingRate:     20 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 5
+			// Sends one 1KB message every 20ms, totaling 50 messages (50KB/s) per second.
+			// The test runs for 400 messages,
+			// expecting a total transmission time of ~8 seconds.
+			name: "queue capacity = 100, " +
+				"total load = 8 * 50 KB, " +
+				"traffic rate = send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          8 * 50,
+			sendQueueCapacity: 10,
+			messagingRate:     20 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 6
+			// Sends one 1KB message every 10ms, totaling 100 messages (100KB/s) per second.
+			// The test runs for 400 messages,
+			// expecting a total transmission time of ~8 seconds.
+			name: "queue capacity = 100, " +
+				"total load = 8 * 50 KB, " +
+				"traffic rate = 2 * send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          8 * 50,
+			sendQueueCapacity: 10,
+			messagingRate:     10 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+		{
+			// testcase 7
+			// Sends one 1KB message every 2ms, totaling 500 messages (500KB/s) per second.
+			// The test runs for 400 messages,
+			// expecting a total transmission time of ~8 seconds.
+			name: "queue capacity = 100, " +
+				"total load = 8 * 50 KB, " +
+				"traffic rate = 10 * send rate",
+			msgSize:           1 * 1024,
+			totalMsg:          8 * 50,
+			sendQueueCapacity: 10,
+			messagingRate:     2 * time.Millisecond,
+			totalDuration:     1 * time.Minute,
+			sendRate:          50 * 1024,
+			recRate:           50 * 1024,
+		},
+	}
+
+	for _, tt := range tests {
+		b.Run(tt.name, func(b *testing.B) {
+			for n := 0; n < b.N; n++ {
+				// set up two networked connections
+				// server, client := NetPipe() // can alternatively use this and comment out the line below
```

Comment:
[optional] if the line below works, do we want to remove this commented code?

```suggestion
```

Reply:
I'd retain this commented part in the test because `NetPipe()` is what is used  throughout all the tests in the `mconnection_test` so this might raise a question for someone reading the test as to why `NetPipe()` is not used here. Therefore, I wanted to clarify this point in place.

Reply:
Makes sense, thanks! Can resolve.